src/audio/dai-zephyr.c

```diff
@@ -553,7 +553,7 @@ static int dai_verify_params(struct dai_data *dd, struct comp_dev *dev,
 		return -EINVAL;
 	}
 
-	if (hw_params.channels && hw_params.channels != params->channels) {
+	if (hw_params.channels && hw_params.channels < params->channels) {
```

Comment:
Hmm, with some DAI implementations (like Intel PCH dmic), the channel count really is fixed, so the old check is correct. But it's true if this constrained cannot be assume to hold for all DAIs, we can enforce the condition here. DAI specific checks needs to return failure later.

Reply:
@dnikodem might be worth an inline comment here to explain. Also it might be worth a 
```c
switch (dai_type) {
case dynamic_dai:
   // compare using new flow
    break;
default:
  // compare using old flow for fixed dais.
   break;
}
```

Reply:
Colleagues, after reading your comments, I'm not sure if my change is correct – I'm working on confirming my doubts. Thanks for the feedback